src/core/predicate/base.rs

```diff
@@ -1,6 +1,10 @@
-use crate::core::{Handler, HandlerResult, TryFromInput};
+use crate::core::{
```

Comment:
I think content of this file must be in mod.rs

Reply:
I prefer to use base module and leave mod.rs for module declarations and reexports.